sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects.scala

```diff
@@ -617,3 +617,31 @@ case class DecodeUsingSerializer[T](child: Expression, tag: ClassTag[T], kryo: B
 
   override def dataType: DataType = ObjectType(tag.runtimeClass)
 }
+
+case class InitializeJavaBean(n: NewInstance, fields: Seq[Expression], setters: Seq[String])
```

Comment:
Also, I would not make it take a `NewInstance` as it would be nice to use a `Literal` sometimes for object reuse.